src/java.base/share/classes/java/lang/reflect/code/op/CoreOps.java

```diff
@@ -1066,8 +1066,21 @@ static Object processConstantValue(TypeDesc t, Object value) {
                 return value == NULL_ATTRIBUTE_VALUE ? null :
                         value.toString();
             } else if (t.equals(TypeDesc.J_L_CLASS)) {
-                return value == NULL_ATTRIBUTE_VALUE ? null :
-                        TypeDesc.ofNominalDescriptor(ClassDesc.of(value.toString()));
+                if (value == NULL_ATTRIBUTE_VALUE) {
+                    return null;
+                }
+                String asString = value.toString();
+                int bracketOpen = asString.indexOf('[');
+                int dimensions = 0;
+                if (bracketOpen != -1) {
+                    dimensions = (asString.length() - bracketOpen) / 2;
+                    asString = asString.substring(0, bracketOpen);
+                }
+                ClassDesc desc = ClassDesc.of(asString);
+                if (dimensions > 0) {
+                    desc = desc.arrayType(dimensions);
+                }
```

Comment:
Instead we can just use `TypeDesc.ofString`.

Until recently we could not parse descriptors in classes in`java.base` that are copied and repackaged in `jdk.compiler` (we are trying to limit the amount of code that is copied). With some refactoring this is now possible for such classes in `java.base` and when copied it is made partial.

Reply:
Ah, that's exactly what I was looking for, thanks!